packages/models/src/resource-binding/manager.ts

```diff
@@ -0,0 +1,201 @@
+import { Transaction, Block, Header, Input, HexString, utils, Cell } from '@ckb-lumos/base'
+import { BI } from '@ckb-lumos/bi'
+import { Actor, ActorMessage, ActorURI, MessagePayload } from '..'
+import { TypeScriptHash, LockScriptHash } from './types'
+import { ResourceBindingRegistry, ResourceBindingManagerMessage } from './interface'
+import { outPointToOutPointString } from './utils'
+import { Listener } from '@ckb-js/kuai-io'
+import type { Subscription } from 'rxjs'
+import { ChainSource } from '@ckb-js/kuai-io/lib/types'
+import { OutPointString } from '../store'
+
+export class Manager extends Actor<object, MessagePayload<ResourceBindingManagerMessage>> {
+  #registry: Map<TypeScriptHash, Map<LockScriptHash, ResourceBindingRegistry>> = new Map()
+  #registryOutpoint: Map<OutPointString, ResourceBindingRegistry> = new Map()
+  #registryReverse: Map<ActorURI, [TypeScriptHash, LockScriptHash]> = new Map()
+  #lastBlock: Block | undefined = undefined
+  #tipBlockNumber = BI.from(0)
+
+  constructor(private _listener: Listener<Header>, private _dataSource: ChainSource) {
+    super()
+  }
+
+  onListenBlock = (blockHeader: Header) => {
+    const currentBlockNumber = BI.from(blockHeader.number)
+    if (currentBlockNumber.gt(this.tipBlockNumber)) {
+      this.tipBlockNumber = currentBlockNumber
+    }
+  }
+
+  private update(pollingInterval = 1000) {
+    return setInterval(async () => {
+      if (this.#tipBlockNumber.gt(0) && this.#tipBlockNumber.gt(BI.from(this.#lastBlock?.header.number ?? 0))) {
+        this.updateStore(await this._dataSource.getBlock(this.#tipBlockNumber.toHexString()))
+      }
+    }, pollingInterval)
+  }
+
+  private updateStore(block: Block) {
+    if (!this.#lastBlock || BI.from(block.header.number).gt(BI.from(this.#lastBlock.header.number))) {
+      const changes = this.filterCells(block)
+      for (const [store, input, data] of changes) {
+        const upgrade: { witness: HexString; cell: Cell }[] = []
+        for (const [cell, witness] of data) {
+          upgrade.push({ witness, cell })
+          const outPoint = cell.outPoint
+          if (outPoint) {
+            this.#registryOutpoint.set(outPointToOutPointString(outPoint), store)
+          }
+        }
+        this.sendMessage(store, 'update_cells', upgrade)
+
+        this.sendMessage(
+          store,
+          'remove_cell',
+          input.map((v) => v.previousOutput),
+        )
+      }
+      this.#lastBlock = block
+    }
+  }
+
+  private filterOutputs(tx: Transaction, block: Block): Map<OutPointString, [Cell, string]> {
```

Comment:
What does the `filter` mean in this method, the logic seems to be an iteration to return all outputs of a transaction

Reply:
I will change the name to "mapOutputs"